back-end/src/sciler/config/checkConfig.go

```diff
@@ -0,0 +1,367 @@
+	}
+	return errorList
+}
+
+// checkActionInstructionType checks if the type op the value of an instruction is the same as the type the instruction requires according to the config
+func checkActionInstructionType(valueType reflect.Kind, instructionType string, value interface{}) error {
+	switch instructionType {
+	case "string":
+		if valueType != reflect.String {
+			return fmt.Errorf("instruction type string expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	case "boolean":
+		if valueType != reflect.Bool {
+			return fmt.Errorf("instruction type boolean expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	case "numeric":
+		if valueType != reflect.Int && valueType != reflect.Float64 {
+			return fmt.Errorf("instruction type numeric expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	case "array":
+		if valueType != reflect.Slice {
+			return fmt.Errorf("instruction type array/slice expected but %s found as type of value %v",
+				valueType.String(), value)
+		}
+	default:
+		return fmt.Errorf("custom types like: %s, are not yet implemented", instructionType)
+	}
+	return nil
+}
+
+// checkActionLabel checks if there is a label with this ID,
+// and checks if all components under this label have the correct instructions with a call to checkActionDevice
+// if the config does not follow the manual, a non-empty list of mistakes is returned
+func checkActionLabel(action Action, config WorkingConfig) []string {
+	errorList := make([]string, 0)
+	if _, ok := config.LabelMap[action.TypeID]; ok { // checks if label can be found in the map, if so, it is stored in variable device
+		for _, instruction := range action.Message {
+			for _, comp := range config.LabelMap[action.TypeID] {
+				instruction.ComponentID = comp.ID
+				errorList = append(errorList,
+					checkActionDevice(Action{TypeID: comp.Device.ID, Message: []ComponentInstruction{instruction}}, config)...)
+			}
+		}
+	} else {
+		errorList = append(errorList,
+			fmt.Sprintf("label with id %s not found in map", action.TypeID))
+	}
+	return errorList
+}
+
+// checkValidComparison checks if the comparison is a valid one
+func checkValidComparison(comparison string) bool {
+	comparisonTypesAllowed := []string{"eq", "lt", "gt", "lte", "gte", "contains", "not"}
+	for _, comp := range comparisonTypesAllowed {
+		if comp == comparison {
+			return true
+		}
+	}
+	return false
+}
+
+// checkConstraints is a method that checks types and comparator operators
+func (or OrCondition) checkConstraints(config WorkingConfig, ruleID string) []string {
+	errorList := make([]string, 0)
+	for _, logic := range or.logics {
+		err := logic.checkConstraints(config, ruleID)
+		if err != nil {
+			errorList = append(errorList, err...)
+		}
+	}
+	return errorList
+}
+
+// checkConstraints is a method that checks types and comparator operators
```

Comment:
checkConstraints is a method that checks types and comparator operators =>
checkConstraints is a method that runs through the constrains under an and condition list and calls a checkContraints on it.